lib/Dialect/MSFT/MSFTOps.cpp

```diff
@@ -56,5 +49,17 @@ void InstanceOp::getAsmResultNames(OpAsmSetValueNameFn setNameFn) {
   }
 }
 
+static LogicalResult verifyInstanceOp(InstanceOp op) {
+  auto topLevelModuleOp = op->getParentOfType<ModuleOp>();
+  StringRef moduleName = op.moduleName();
+  Operation *referencedModule = topLevelModuleOp.lookupSymbol(moduleName);
```

Comment:
This is an extremely expensive O(n) lookup.  I'd recommend moving the MSFT dialect to conform to the `SymbolUserOpInterface` trait like HWInstance does, and verify the instance/module link in the same way.

Reply:
If you're OK with it, I'll open an issue for this. I'm only looking to address #1779 in this PR. Having this verification at all is a step in the right direction.

Reply:
Yeah, makes perfect sense to split that out as a separate issue/PR.

Reply:
This was simple enough; SymbolUserOpInterface is added and used instead of the lookupSymbol.